### PR TITLE
Moar Py3

### PR DESCRIPTION
**Pull request by s0undt3ch**

### CODE REVIEW

**Comment by s0undt3ch** [comment]
I really don't know what's up with that docker test. @terminalmage, any insights?

**Comment by s0undt3ch** [comment]
Unrelated failure. Merging.

File: salt/loader.py [hunk 1]
```diff
@@ -1177,10 +1177,10 @@ def refresh_file_mapping(self):
         # The files are added in order of priority, so order *must* be retained.
         self.file_mapping = salt.utils.odict.OrderedDict()
 
-        for mod_dir in self.module_dirs:
+        for mod_dir in sorted(self.module_dirs):
```
Comment (ScoreUnder) [inline salt/loader.py]: I know it's a little late now that this is merged, but this looks like a mistake to me. The `module_dirs` field is populated in order of priority, so the sorting will break this.

There are comments in the `_module_dirs` functions like this one:
```
        # Prepend to the list to match cli argument ordering
```
Which suggest that the author does not expect the list to be sorted afterwards.

Comment (s0undt3ch) [inline salt/loader.py]: I created https://github.com/saltstack/salt/issues/40517 in order not to forget about this

**Pull request merged**